save-plugins/warn-plugin/src/commonMain/kotlin/org/cqfn/save/plugin/warn/WarnPlugin.kt

```diff
@@ -92,20 +92,17 @@ class WarnPlugin : Plugin {
                                        warnPluginConfig: WarnPluginConfig): TestStatus {
         val missingWarnings = expectedWarningsMap.filterValues { it !in actualWarningsMap.values }.values
         val unexpectedWarnings = actualWarningsMap.filterValues { it !in expectedWarningsMap.values }.values
-        if (!warnPluginConfig.exactWarningsMatch && missingWarnings.isEmpty() && unexpectedWarnings.isNotEmpty()) {
-            return Pass("Some warnings were unexpected: $unexpectedWarnings")
+
+        val isMissingWarnings = missingWarnings.isEmpty()
+        val isUnexpectedWarnings = unexpectedWarnings.isEmpty()
+        return when {
```

Comment:
Why did you remove the old version with `when (missingWarnings.isEmpty() to unexpectedWarnings.isEmpty())`?

Reply:
For one case, was needed a third parameter - warnPluginConfig.exactWarningsMatch

Reply:
You can place an additional `if` inside `when`'s branch. For me now it looks less clear, with multiple repeated variable names

Reply:
Ok. Fixed it